components/emoji_picker/modules/emoji_selector.vue

```diff
@@ -0,0 +1,546 @@
+<template>
+  <div class="d-emoji-picker__selector">
+    <div
+      ref="listRef"
+      class="d-emoji-picker__list"
+    >
+      <p
+        v-if="emojiFilter"
+        class="d-emoji-picker__search-label d-emoji-picker__alignment"
+      >
+        {{ filteredEmojis.length > 0 ? searchResultsLabel : searchNoResultsLabel }}
+      </p>
+      <div
+        v-else
+        ref="tabCategoryRef"
+        class="d-emoji-picker__category d-emoji-picker__alignment"
+      >
+        <p>
+          {{ fixedLabel }}
+        </p>
+      </div>
+      <div
+        v-for="(tabLabel, index) in tabLabels"
+        v-show="!emojiFilter"
+        :key="index"
+        :ref="tabLabel.ref"
+        class="d-emoji-picker__alignment"
+      >
+        <p
+          v-if="index"
+        >
+          {{ tabLabel.label }}
+        </p>
+        <div
+          class="d-emoji-picker__tab"
+        >
+          <button
+            v-for="emoji in (emojis[tabs[index] + skinTone] ? emojis[tabs[index] + skinTone] : emojis[tabs[index]])"
+            :key="emoji.shortname"
+            type="button"
+            :aria-label="emoji.name"
+            @click="$emit('selected-emoji', emoji)"
+            @focusin="$emit('emoji-data', emoji)"
```

Comment:
I find `emoji-data` kind of generic for an event name. does something like `highlighted-emoji` work?

Reply:
Seems legit to me, will update it